binaries/s3_binary_upload.py

```diff
@@ -0,0 +1,108 @@
+import argparse
+import glob
+import logging
+import os
+import subprocess
+import sys
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(logging.INFO)
+LOGGER.addHandler(logging.StreamHandler(sys.stderr))
+
+
+REPO_ROOT = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..")
+
+TS_WHEEL_PATH = glob.glob(os.path.join(REPO_ROOT, "dist"))[0]
+MA_WHEEL_PATH = glob.glob(os.path.join(REPO_ROOT, "model-archiver", "dist"))[0]
+WA_WHEEL_PATH = glob.glob(os.path.join(REPO_ROOT, "workflow-archiver", "dist"))[0]
+
+
+class S3BinaryUploader:
+    def __init__(self, s3_bucket: str, is_dryrun: bool = False, is_nightly: bool = False):
+        """
+        Initialize the uploader with s3_bucket, and s3_backup_bucket
+        """
+        self.s3_bucket = s3_bucket
+        self.dryrun = is_dryrun
+        if self.dryrun:
+            self.s3_command = "aws s3 cp --recursive --dryrun"
+        else:
+            self.s3_command = "aws s3 cp --recursive"
+
+        self.channel = "nightly" if is_nightly else ""
+
+    def s3_upload_default_binaries(self):
+        """
+        Uploads the *.whl files provided in the standard directory structure of the pytorch 'serve' directory,
+        assuming that the *.whl files are available in the 'dist' folder of the 
+        is_backup: bool: boolean value that specified whether or not to upload the binaries to a backup S3 bucket
+        """
+        for local_folder_path in [TS_WHEEL_PATH, MA_WHEEL_PATH, WA_WHEEL_PATH]:
+            s3_command = f"{self.s3_command} --exclude '*' --include '*.whl' {local_folder_path} {self.s3_bucket.rstrip('/')}/whl/{self.channel}"
+
+            try:
+                ret_code = subprocess.run(
+                    s3_command, check=True, stdout=subprocess.PIPE, universal_newlines=True, shell=True
+                )
+
+            except subprocess.CalledProcessError as e:
+                LOGGER.info(f"S3 upload command failed: {s3_command}. Exception: {e}")
+
+            LOGGER.info(f"S3 upload successful using command: {s3_command}")
+
+    def s3_upload_local_folder(self, local_folder_path: str):
```

Comment:
it is better to refactor the code to reduce code duplication in  s3_upload_default_binaries and s3_upload_local_folder.

Reply:
Done, retained the function, as a caller function to the other function.